spec/steps/discover.fmf

```diff
@@ -1,44 +1,99 @@
-summary: Discover test relevant for execution
+summary: Discover tests relevant for execution
 
 description: |
-    Gather information about the test cases which are supposed to
-    be run. This includes list of test cases with corresponding
-    L1 metadata. From the test case metadata constraints for test
-    enviroment can be detected:
+    Gather information about tests which are supposed to be run.
+    Provide method ``tests()`` returning a list of discovered
+    tests and ``requires()`` returning a list of all required
+    packages gathered from individual test metadata.
 
-    * Architectures supported
-    * Disk and memory constraints
-    * Product relevancy
-    * Environment variables
-
-    Examples of metadata storage:
-
-    shell
-        Manual list of shell test cases
-    fmf
-        Flexible Metadata Format filter
-
-    For each ``repository`` the discover step should produce a
-    ``repository.yaml`` file with the list of tests in the
-    following format::
+    Store the list of aggregated tests with their corresponding
+    metadata in the ``tests.yaml`` file. The format should be a
+    dictionary of dictionaries structured in the following way::
 
         /test/one:
-            test: ./test.sh
-            path: /test/path
-            duration: 5m
+            summary: Short test summary.
+            description: Long test description.
+            contact: Petr Šplíchal <psplicha@redhat.com>
+            component: [tmt]
+            test: tmt --help
+            path: /test/path/
+            require: [required, packages]
```

Comment:
`[package1, package2]` would be less misleading maybe?

Reply:
Hm, makes sense. Updated.